pkg/api/types.go

```diff
@@ -623,6 +629,30 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// Represents a Quobyte mount that lasts the lifetime of a pod.
+// Quobyte volumes do not support ownership management or SELinux relabeling.
+type QuobyteVolumeSource struct {
+	// Registry represents a single or multiple Quobyte Registry services
+	// specified as a string as host:port pair (multiple entries are separated with commas)
+	// which acts as the central registry for volumes
+	Registry string `json:"registry"`
+
+	// Volume is a string that references an already created Quobyte volume by name.
+	Volume string `json:"volume"`
+
+	// Defaults to false (read/write). ReadOnly here will force
+	// the Quobyte to be mounted with read-only permissions
+	ReadOnly bool `json:"readOnly,omitempty"`
+
+	// User to map volume access to
+	// Defaults to the root user
+	User string `json:"user,omitempty"`
```

Comment:
@saad-ali @pmorie 

With the Quobyte volume driver, we are allowing the user to specify which (Unix) user and group a container uses to access data. This allows us to do fine-grained access control in file systems that are shared between applications and users, and makes an application deployment entirely independent from the (random) unix user baked into the container.

We eventually want to hook that up to the existing service account concept, so that it defaults to the service account name.